src/client/features/view/components/tooltips/generateContent.js

```diff
@@ -0,0 +1,238 @@
+}
+
+//Render a standard tooltip item
+//Strings -> HTML
+function makeTooltipItem(value, field) {
+  return h('div.fake-paragraph', [h('div.field-name', field), h('div.tooltip-value', value.toString())]);
+}
+
+//Delete duplicates and ignore case
+//Requires a valid array
+//Array -> Array 
+function deleteDuplicatesWithoutCase(list) {
+  if (!(list instanceof Array)) {
+    return list;
+  }
+  return list.reduce((result, element) => {
+    var normalize = function (x) { return typeof x === 'string' ? x.toLowerCase() : x; };
+    var normalizedElement = normalize(element);
+    if (result.every(otherElement => normalize(otherElement) !== normalizedElement)) {
+      result.push(element);
+    }
+    return result;
+  }, []);
+}
+
+
+//Generate a data source link based on database name
+//Requires a valid database name
+//Note : prefix is optional
+//String -> HTML 
+function generateDataSourceLink(name, prefix = '') {
+  let db = config.databases;
+  let link = db.filter(value => name.toUpperCase().indexOf(value[0].toUpperCase()) !== -1);
+  if (link.length === 1 && link[0][1]) {
+    return h('div', h('div.field-name', prefix), h('a.plain-link', { href: link[0][1], target: '_blank' }, link[0][0]));
+  }
+  else if (link.length === 1) {
+    return h('div', h('div.field-name', prefix), h('a.plain-link', link[0][1]));
+  }
+  else {
+    return h('div', h('div.field-name', prefix), h('a.plain-link', name));
+  }
+}
+
+//Sort Database ID's by database name
+//Requires a valid database ID array
+//Array -> Array 
+function sortByDatabaseId(dbArray) {
+  let databases = {};
+
+  //Group Ids by database
+  for (let i = 0; i < dbArray.length; i++) {
+    let databaseName = dbArray[i][0];
+    if (!databases[databaseName]) {
+      databases[databaseName] = [];
+    }
+    databases[databaseName].push(dbArray[i][1]);
+  }
+
+  //Produce final array with group names as the keys
+  dbArray = [];
+  for (let databaseName in databases) {
+    dbArray.push({ database: databaseName, ids: databases[databaseName] });
+  }
+
+  //Return result
+  return dbArray;
+}
+
+//Generate list of all given database id's
+//Requires a valid database Id Object
+//Array -> HTML
+function generateIdList(dbIdObject, trim) {
+  //get name and trim ID list to 5 items
+  let name = dbIdObject.database;
+  let list = dbIdObject.ids;
+  let db = config.databases;
+
+  //Format names
+  let dbScan = db.filter(data => name.toUpperCase().indexOf(data[0].toUpperCase()) !== -1);
+  if (dbScan.length > 0) { name = dbScan[0][0]; }
+
+  //Trim list
+  if (trim) { list = dbIdObject.ids.slice(0, 1); }
+
+  //Generate a list or a single link
+  if (list.length == 1 && trim) {
+    return h('li.db-item', generateDBLink(name, list[0], true));
+  }
+  else {
+    return h('li.db-item', h('div.db-name', name + ": "), list.map(data => generateDBLink(name, data, false), this));
+  }
+}
+
+//Generate a database link
+//Strings -> HTML 
+function generateDBLink(dbName, dbId, printId) {
```

Comment:
printId seems like some magic value to me. I can't understand what this is for.

Reply:
`printId` is actually a boolean and is supposed to indicate if the databaseId is displayed on screen. 

I have renamed the variable to `isIdVisible` to make the purpose and type clearer